frontend/viewer/src/lib/components/reorderer/reorderer-item-list.svelte

```diff
@@ -0,0 +1,62 @@
+<script lang="ts" module>
+  export type ReordererItemListProps = Omit<HTMLAttributes<HTMLElement>, 'onchange'> & {
+    id?: string;
+  };
+</script>
+
+<script lang="ts" generics="T">
+  import {watch} from 'runed';
+  import {Icon} from '$lib/components/ui/icon';
+  import * as DropdownMenu from '$lib/components/ui/dropdown-menu';
+  import type {HTMLAttributes} from 'svelte/elements';
+  import {mergeProps} from 'bits-ui';
+  import {cn} from '$lib/utils';
+  import {useReordererItemList} from './reorderer.svelte';
+
+  let {
+    ...rest
+  } : ReordererItemListProps = $props();
+
+  const itemListState = useReordererItemList<T>();
+  const {item, items, getDisplayName, onchange} = $derived(itemListState);
+
+  const currIndex = $derived(items.indexOf(item));
+  let displayItems = $derived(items);
+  let displayIndex = $derived(currIndex);
+
+  watch([() => items, () => displayIndex], () => {
+    const newDisplayItems = [...items];
+    newDisplayItems.splice(currIndex, 1);
+    newDisplayItems.splice(displayIndex, 0, items[currIndex]);
+    displayItems = newDisplayItems;
```

Comment:
based on this `displayItems` should just be derived by this function callback, no watch needed.